docs/changelog/0.4.0.md

```diff
@@ -0,0 +1,532 @@
+
+# 0.4.0 (Jan 14, 2021)
+
+We are happy to announce the next major release of Scala Native!
+
+Release 0.4.0 brings many changes to Scala Native ecosystem, let's take a look at few of them. 
+
+## Compiler changes
+### Supported Scala version
+The most important change introduced by this release is support for recent Scala versions, that means you're able to work
+with the most recent versions of Scala 2 available at the moment of release - 2.12.12 and 2.13.4. 
+Scala 2.11 is still support, what more it was enhanced by multiple features, i.e. handling of java default methods.
+
+### JUnit Native Plugin
+Scala Native now comes with JUnit support out of the box, this means that you can write tests in the same way you would do for a Java project.
+To enable JUnit tests all you will need to do is to add the two following lines to your `build.sbt`.
+
+```scala
+addCompilerPlugin("org.scala-native" % "junit-plugin" % SNVersion cross CrossVersion.full)
+```
+
+### NIR changes
+With this major release we're forced to break backward compatibility with previous versions of SN. 
+It means also means that any library published using version 0.4.0-M2 or older would not be compatible to use now.
+
+We also have managed to fix multiple Native Intermiediete Representation related bugs, i.e the way Strings and Characters 
+were stored or adding class literal allowing us to handle proper caching of `java.lang.Class` instances. 
+Moreover, we are now collecting source code positions in NIR instructions, allowing us to add first class debug support in following releases.
+
+### Reflective instantiation
```

Comment:
We should mention somewhere in this section the API is exactly equivalent to what Scala.js has.